src/main/java/com/uid2/admin/job/JobDispatcher.java

```diff
@@ -2,39 +2,56 @@
 
 import com.uid2.admin.job.model.Job;
 import com.uid2.admin.job.model.JobInfo;
+import com.uid2.admin.store.Clock;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Metrics;
 import io.vertx.core.logging.Logger;
 import io.vertx.core.logging.LoggerFactory;
 
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
 import java.util.*;
 import java.util.concurrent.*;
 import java.util.stream.Collectors;
 
 public class JobDispatcher {
-    private static class Loader {
-        public static final JobDispatcher INSTANCE = new JobDispatcher();
-    }
-
     private static final Logger LOGGER = LoggerFactory.getLogger(JobDispatcher.class);
 
-    private final ExecutorService JOB_EXECUTOR = Executors.newSingleThreadExecutor();
-    private final Queue<Job> JOB_QUEUE = new ConcurrentLinkedQueue<>();
-    private final Object JOB_LOCK = new Object();
+    private final ExecutorService jobExecutor = Executors.newSingleThreadExecutor();
+    private final Queue<Job> jobQueue = new ConcurrentLinkedQueue<>();
+    private final Object jobLock = new Object();
+
+    private final String id;
+    private final int intervalMs;
+    private final int maxRetries;
+    private final Clock clock;
+
     private boolean started = false;
+    private Job currentJob = null;
     private ScheduledExecutorService scheduler;
-    private Job currentJob;
 
-    private JobDispatcher() {}
-
-    public static JobDispatcher getInstance() {
-        return Loader.INSTANCE;
+    public JobDispatcher(
+            String id,
+            int intervalMs,
+            int maxRetries,
+            Clock clock) {
+        this.id = id;
+        this.intervalMs = intervalMs;
+        this.maxRetries = maxRetries;
+        this.clock = clock;
+
+        Gauge.builder("uid2.job_dispatcher.execution_duration_ms", this::getExecutionTime)
+                .tag("job_dispatcher", id)
+                .description("gauge for " + id + " execution time")
```

Comment:
Don't need to include "gauge for", just like variable names don't need the word "variable" in them :)

Reply:
I think all the other descriptions have the same format so I'm just following those descriptions